cornedbeef/src/main/java/com/swiftkey/cornedbeef/CoachMark.java

```diff
@@ -166,7 +166,12 @@ public void run() {
                         if (mTimeoutListener != null) {
                             mTimeoutListener.onTimeout();
                         }
-                        dismiss();
+                        try {
+                            dismiss();
+                        } catch(IllegalArgumentException e) {
```

Comment:
Can we check whether dismissed or not by like isShowing() or something similar way instead of catching exception?

Reply:
we can but it seems like this isn't fall proof:

https://stackoverflow.com/questions/2745061/java-lang-illegalargumentexception-view-not-attached-to-window-manager

I think if we're running things directly on the Handler we really should be extra cautious to ensure we don't force close apps that implement this library.